docs/_docstrings_parser.py

```diff
@@ -0,0 +1,94 @@
+# type: ignore  # noqa: PGH003
+import re
+
+from docutils import nodes
+from myst_parser.parsers.sphinx_ import MystParser
+
+
+class GoogleStyleDocstringParser(MystParser):
```

Comment:
Do we need this parser if we just use the myst style it is outputting? As that is what puya stubs is using 

e.g. 
https://github.com/algorandfoundation/puya/blob/main/stubs/algopy-stubs/arc4.pyi#L80
becomes
https://algorandfoundation.github.io/puya/api-algopy.arc4.html#algopy.arc4.baremethod

Reply:
@daniel-makerx pushed to state ops to propagate changes outside of docs folder there first, updated google to reST styled docstrings (which i assume are the ones compliant with PEP 257), also added extra step to linting to run https://docformatter.readthedocs.io/en/latest/ from PYCQA -> which autoformats a subset of pep 257 rules. Makes docstrings easier to deal with.

Reply:
Removed google docstring parser but will keep it for reference on local stash - might still get handy for some other algokit repos that use that format of reST